packages/p2p/stores/my-profile-store.js

```diff
@@ -0,0 +1,159 @@
+import { observable, action } from 'mobx';
+import { requestWS } from 'Utils/websocket';
+import { localize } from 'Components/i18next';
+import { textValidator } from 'Utils/validations';
+
+export default class MyProfileStore {
+    @observable advertiser_info = {};
+    @observable contact_info = '';
+    @observable default_advert_description = '';
+    @observable error_message = '';
+    @observable form_error = '';
+    @observable is_button_loading = false;
+    @observable is_loading = true;
+    @observable is_submit_success = false;
+    @observable payment_info = '';
+
+    @action.bound
+    getAdvertiserInfo() {
+        this.setIsLoading(true);
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_info: 1,
+            }).then(response => {
+                if (!response.error) {
+                    const { p2p_advertiser_info } = response;
+                    this.setAdvertiserInfo(p2p_advertiser_info);
+                    this.setContactInfo(p2p_advertiser_info.contact_info);
+                    this.setDefaultAdvertDescription(p2p_advertiser_info.default_advert_description);
+                    this.setPaymentInfo(p2p_advertiser_info.payment_info);
+                } else {
+                    this.setErrorMessage(response.error);
+                }
+                this.setIsLoading(false);
+                resolve();
+            });
+        });
+    }
+
+    @action.bound
+    handleSubmit(values) {
+        this.setIsButtonLoading(true);
+
+        return new Promise(resolve => {
```

Comment:
Same here, does this still need to be a promise?

Reply:
@aaron-binary I've merged functions, removed promise and left props for now, as discussed. Thank you! :)